python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -788,7 +768,8 @@ def _flush(self, _objs=None):
             for obj_id in obj_ids:
                 self.unlock_transaction(obj_id)
 
-            self._lock.release()
+            for obj in objs:
+                obj._releaseParentLock()
```

Comment:
This should never happen but I've seen enough cases, of 'should not happen' problems at the registry level that I'm wanting to keep the locks around repo access in the registry as strict as possible.